lib/cstruct.ml

```diff
@@ -96,7 +96,7 @@ let create_unsafe len =
   { buffer ; len ; off = 0 }
 
 let check_bounds t len =
-  Bigarray.Array1.dim t.buffer >= len
+  len >= 0 && Bigarray.Array1.dim t.buffer >= len
```

Comment:
Yes, `Invalid_argument` would be reasonable.  I'm not sure how the public function is intended to be used: it uses the size of the underlying buffer, not the length of the cstruct itself, which is a bit different from most functions in the API.

It'd be quite handy to have arithmetic operations that raise on overflow in the standard library.